### PR TITLE
use ruff as linter

### LINKED ISSUE #217: [Bug]: Pydocstyle is deprecated. Switch on ruff.
### Severity

Medium

### What release version, tag or commit-hash did you use?

c74e63abc9e5d6aa2ee0d69159c7c3733465c144

### Current Behavior

Deprecated pydocstyle is in use.

### Steps to Reproduce

---

### Expected Behavior

Switch on ruff

### Possible Solution

Switch on ruff

### Additional Information

_No response_

### Code of Conduct

- [X] I agree to follow this project's "Code of Conduct".

### PR DESCRIPTION
**Pull request by MP91**
# Description
fixes #217, #193 and #184 

## Checklist

* [ ] Vehicle App can be started with dapr run and is connecting to vehicle data broker
* [ ] Vehicle App can process MQTT messages and call the seat service
* [ ] Vehicle App can be deployed to local K3D and is running
* [ ] Created/updated tests, if necessary. Code Coverage percentage on new code shall be >= 70%.
* [ ] Extended the documentation in Velocitas repo
* [ ] Extended the documentation in README.md

* [ ] Devcontainer can be opened successfully
* [ ] Devcontainer can be opened successfully behind a corporate proxy
* [ ] Devcontainer can be re-built successfully

* [ ] Release workflow is passing

### CODE REVIEW

File: .velocitas.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
         },
         {
             "name": "devenv-devcontainer-setup",
-            "version": "v1.4.5"
```
Comment (MP91) [inline .velocitas.json:17]: use correct tag

File: NOTICE.md [hunk 2]
```diff
@@ -21,12 +21,15 @@ SPDX-License-Identifier: Apache-2.0
...
-| Mosquitto           | https://github.com/eclipse/mosquitto/blob/master/LICENSE.txt                                                                                                                 |
+| Mosquitto           | https://github.com/eclipse/mosquitto/blob/master/LICENSE.txt                                                     |
 
 ### Development
-| Software |                               License                                |
-| :------: | :------------------------------------------------------------------: |
-|   GRPC   |           https://github.com/grpc/grpc/blob/master/LICENSE           |
+| Software   |                               License                                |
+| :------:   | :------------------------------------------------------------------: |
+|   GRPC     |           https://github.com/grpc/grpc/blob/master/LICENSE           |
```
Comment (doosuu) [inline NOTICE.md:29]: Fix alignment and spelling on gRPC.

File: setup.cfg
Comment (doosuu) [inline setup.cfg:1]: Let's create a follow-up issue to switch to pyproject.toml?

**Review by doosuu — CHANGES_REQUESTED** [review]

File: setup.cfg
Comment (doosuu) [reply setup.cfg:1]: Not specific to ruff - pyproject.toml is the new de-facto standard. While setup.py is not deprecated, I would recommend we switch at some point.

File: .velocitas.json (hunk 1 above)
Comment (MP91) [reply .velocitas.json:17]: done

File: NOTICE.md (hunk 2 above)
Comment (MP91) [reply NOTICE.md]: done

**Review by doosuu — APPROVED** [review]
lgtm

**Pull request merged**